mycroft/skills/main.py

```diff
@@ -17,45 +17,99 @@
 
 
 import json
-
 import sys
+import time
+from threading import Timer
+
+import os
 from os.path import expanduser, exists
 
 from mycroft.configuration import ConfigurationManager
 from mycroft.messagebus.client.ws import WebsocketClient
-from mycroft.skills.core import load_skills, THIRD_PARTY_SKILLS_DIR
+from mycroft.skills.core import load_skills, THIRD_PARTY_SKILLS_DIR, \
+    load_skill, create_skill_descriptor, MainModule
 from mycroft.util.log import getLogger
 
 logger = getLogger("Skills")
 
 __author__ = 'seanfitz'
 
 ws = None
-skills = []
+loaded_skills = {}
+last_modified_skill = 0
+skills_directories = []
 
 
-def load_skills_callback():
+def connect():
     global ws
-    global skills
-    skills += load_skills(ws)
-    config = ConfigurationManager.get().get("skills")
+    ws.run_forever()
+
+
+def load_watch_skills():
+    global ws, loaded_skills, last_modified_skill, skills_directories
+
+    skills_directories = [os.path.dirname(os.path.abspath(__file__))]
+    skills_directories = skills_directories + THIRD_PARTY_SKILLS_DIR
 
     try:
+        config = ConfigurationManager.get().get("skills")
         ini_third_party_skills_dir = expanduser(config.get("directory"))
+        if ini_third_party_skills_dir and exists(ini_third_party_skills_dir):
+            skills_directories.append(ini_third_party_skills_dir)
     except AttributeError as e:
         logger.warning(e.message)
 
-    for loc in THIRD_PARTY_SKILLS_DIR:
-        if exists(loc):
-            skills += load_skills(ws, loc)
+    timer = Timer(0, load_skills)
+    timer.daemon = True
+    timer.start()
 
-    if ini_third_party_skills_dir and exists(ini_third_party_skills_dir):
-        skills += load_skills(ws, ini_third_party_skills_dir)
+    # last_modified_skill = max(skills.values())
 
 
-def connect():
+def clear_skill_events(instance):
     global ws
-    ws.run_forever()
+    events = ws.emitter._events
+    instance_events = []
+    for event in events:
+        e = ws.emitter._events[event]
+        if len(e) > 0 and e[0].func_closure and isinstance(
+                e[0].func_closure[1].cell_contents, instance.__class__):
+            instance_events.append(event)
+
+    for event in instance_events:
+        del events[event]
+
+
+def load_skills():
```

Comment:
can we name this something else? We already have load_skills and load_skill in core.py